File: 2025-1/sesion-de-clase-15/SoftProgNegocio/BOImpl/AreaBOImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PUCP.Edu.Pe.SoftProg.Modelo.RRHH;
using PUCP.Edu.Pe.SoftProg.Negocio.BO;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAO.RRHH;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAOImpl.RRHH;

namespace PUCP.Edu.Pe.SoftProg.Negocio.BOImpl {
    public class AreaBOImpl : IAreaBO {
        private readonly IAreaDAO areaDAO;

        public AreaBOImpl() { 
            this.areaDAO = new AreaDAOImpl();
        }

        public void Eliminar(int id) {
            if (!this.areaDAO.Eliminar(id)) {
                throw new Exception("No se pudo eliminar el area");
            }
        }

        public void Guardar(Area area, Estado estado) {
            if (estado == Estado.Nuevo) {
                this.areaDAO.Insertar(area);
            }
            else if (estado == Estado.Modificar) { 
                this.areaDAO.Modificar(area);
            }
        }

        public List<Area> Listar() {
            return this.areaDAO.Listar();
        }

        public Area Obtener(int id) {
            return this.areaDAO.Buscar(id);
        }
    }
}

File: 2025-1/sesion-de-clase-18/SoftProg/SoftProgNegocio/src/main/java/pe/edu/pucp/inf30/softprog/boimpl/AreaBOImpl.java
package pe.edu.pucp.inf30.softprog.boimpl;

import java.util.List;
import pe.edu.pucp.inf30.softprog.bo.Estado;
import pe.edu.pucp.inf30.softprog.bo.IAreaBO;
import pe.edu.pucp.inf30.softprog.dao.rrhh.IAreaDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.rrhh.AreaDAOImpl;
import pe.edu.pucp.inf30.softprog.model.rrhh.Area;

/**
 *
 * @author eric
 */
public class AreaBOImpl implements IAreaBO {
    private final IAreaDAO areaDao;
    
    public AreaBOImpl() {
        this.areaDao = new AreaDAOImpl();
    }
    
    @Override
    public List<Area> listar() {
        return this.areaDao.listar();
    }

    @Override
    public Area obtener(int id) {
        return this.areaDao.buscar(id);
    }

    @Override
    public void eliminar(int id) {
        this.areaDao.eliminar(id);
    }

    @Override
    public void guardar(Area modelo, Estado estado) {
        if (estado == Estado.Nuevo) {
            this.areaDao.insertar(modelo);
        }
        else {
            this.areaDao.modificar(modelo);
        }
    }   
}